src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,302 @@
+                    if (index < tasks.getTasks().size()) {
+                        tasks.markTaskNotDone(index);
+                        ui.sayMessage("Nice! I've marked this task as not done:\n" + tasks.getTask(index).toString());
+                    } else {
+                        throw new DukeException("No such task exists.");
+                    }
+                    storage.OverwriteFile(tasks.getTasks());
+                    break;
+                }
+                case DELETE:
+                    int index = parser.parseIndex(input);
+                    if (index < tasks.getTasks().size()) {
+                        Task removedTask = tasks.removeTask(index);
+                        ui.sayMessage("Noted. I've removed this task:\n" + removedTask.toString());
+                    } else {
+                        throw new DukeException("No such task exists.");
+                    }
+                    storage.OverwriteFile(tasks.getTasks());
+                    break;
+                case TODO: {
+                    Todo newTodo = new Todo(parser.parseTodo(input));
+                    tasks.addTask(newTodo);
+                    ui.sayMessage(
+                            "Got it. I've added this task:\n" + newTodo + "\n" + "Now you have " + tasks.getTasks()
+                                    .size() + " tasks in the list.");
+                    storage.OverwriteFile(tasks.getTasks());
+                    break;
+                }
+                case DEADLINE:
+                    String[] deadlineInput = parser.parseDeadline(input);
+                    Deadline newDeadline = new Deadline(deadlineInput[0], deadlineInput[1]);
+                    tasks.addTask(newDeadline);
+                    ui.sayMessage(
+                            "Got it. I've added this task:\n" + newDeadline + "\n" + "Now you have " + tasks.getTasks()
+                                    .size() + " tasks in the list.");
+                    storage.OverwriteFile(tasks.getTasks());
+                    break;
+                case EVENT:
+                    String[] eventInput = parser.parseEvent(input);
+                    Event newEvent = new Event(eventInput[0], eventInput[1]);
+                    tasks.addTask(newEvent);
+                    ui.sayMessage(
+                            "Got it. I've added this task:\n" + newEvent + "\n" + "Now you have " + tasks.getTasks()
+                                    .size() + " tasks in the list.");
+                    storage.OverwriteFile(tasks.getTasks());
+                    break;
+                case FIND:
+                    String word = parser.parseFind(input);
+                    ui.listTasks(tasks.findTasks(word));
+                    break;
+                case ERROR:
+                    throw new DukeException("Sorry :( I don't know what this means.");
+                }
+            } catch (DukeException e) {
+                ui.sayMessage(e.getErrorMsg());
```

Comment:
would it be better to handle this cases somewhere else or create another class to avoid populating the duke class?

Reply:
Yeah I will probably encapsulate the functionality in another class soon.